scripts/build.py

```diff
@@ -1026,17 +1059,23 @@ def _verify_build(input_dirnames, output_dirnames, file_hashes):
         hashes are inserted.
 
         Args:
-            input_dirnames: list(str). List of directory paths that contain
-                source files.
+            input_dirnames: list(str|list(str)). List of directory paths that
+                contain source files.
             output_dirnames: list(str). List of directory paths that contain
                 built files.
+            input_dirnames_type: Type of input_dirnames - whether it is a single
```

Comment:
No, don't do this. Find a way to break things down that makes things clearer (see example above), don't do type magic via additional args. Having to do if/else for stuff like this is an antipattern and makes the code harder to maintain.

Reply:
Updated.